pkg/login/social/generic_oauth_test.go

```diff
@@ -427,3 +427,68 @@ func TestUserInfoSearchesForLogin(t *testing.T) {
 		}
 	})
 }
+
+func TestPayloadCompression(t *testing.T) {
+	t.Run("Given a generic OAuth provider", func(t *testing.T) {
+		provider := SocialGenericOAuth{
+			SocialBase: &SocialBase{
+				log: log.NewWithLevel("generic_oauth_test", log15.LvlDebug),
+			},
+			emailAttributePath: "email",
+		}
+
+		tests := []struct {
+			Name          string
+			OAuth2Extra   interface{}
+			ExpectResult  bool
```

Comment:
I think this attribute is superfluous and it's enough to use the default value for `ExpectedEmail` to signal that there should be no result.

Reply:
This test is calling provider.extractFromToken(token), which returns nil on error.... so I can't check the userInfo.Email value.  I tried switching the test to make an ExpectedEmail == "" imply that userInfo should be nil, but that didn't really seem clearer that the way it is.